src/pke/include/schemerns/rns-cryptoparameters.h

```diff
@@ -827,6 +821,9 @@ class CryptoParametersRNS : public CryptoParametersRLWE<DCRTPoly> {
 
   const NativeInteger &GetScalingFactorInt(usint l) const {
     if (m_rsTechnique == FLEXIBLEAUTO) {
```

Comment:
I guess the code would be a bit more readable in this form: 
  double GetScalingFactorReal(uint32_t l = 0) const {
    if (m_rsTechnique == FLEXIBLEAUTO && l < m_scalingFactorsReal.size()) {
      return m_scalingFactorsReal[l];
    }

    return m_approxSF;
  }